packages/optimizely-sdk/lib/utils/semantic_version/index.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright 2020, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getLogger } from '@optimizely/js-sdk-logging';
+import { VERSION_TYPE, LOG_MESSAGES } from '../enums';
+
+const MODULE_NAME = 'SEMANTIC VERSION';
+const logger = getLogger();
+
+/**
+  * Evaluate if provided string is number only
+  * @param  {unknown}  content
+  * @return {boolean}  true if the string is number only
+  *                     
+  */
+  function isNumber(content: string): boolean {
+    return content.match(/^[0-9]+$/) != null ? true : false;
+  }
+
+ /**
+  * Evaluate if provided version contains pre-release "-"
+  * @param  {unknown}  version
+  * @return {boolean}  true if the version contains "-" and meets condition
+  *                     
+  */
+  function isPreReleaseVersion(version: string): boolean {
+    const preReleaseIndex = version.indexOf(VERSION_TYPE.IS_PRE_RELEASE);
+    const buildIndex = version.indexOf(VERSION_TYPE.IS_BUILD);
+
+    if (preReleaseIndex < 0) {
+        return false;
+    }
+
+    if (buildIndex < 0 ) {
+        return true;
+    }
+
+    return preReleaseIndex < buildIndex;
+  }
+  
+  /**
+   * Evaluate if provided version contains build "+"
+   * @param  {unknown}  version
+   * @return {boolean}  true if the version contains "+" and meets condition
+   *                     
+   */
+  function isBuildVersion(version: string): boolean {
+    const preReleaseIndex = version.indexOf(VERSION_TYPE.IS_PRE_RELEASE);
+    const buildIndex = version.indexOf(VERSION_TYPE.IS_BUILD);
+
+    if (buildIndex < 0) {
+        return false;
+    }
+
+    if (preReleaseIndex < 0 ) {
+        return true;
+    }
+
+    return buildIndex < preReleaseIndex;
+  }
+  
+  /**
+   * check if there is any white spaces " " in version
+   * @param  {unknown}  version
+   * @return {boolean}  true if the version contains " "
+   *                     
+   */
+  function hasWhiteSpaces(version: string): boolean {
+    return version.includes(' ');
```

Comment:
What about other space characters (tab, newline, etc.)? Should we use space regexp character class?
```suggestion
    return /\s/.test(version);
```
- Minor concern: `String.prototype.includes` is not supported in very old JS environments.
- Is there any performance difference between `String.prototype.includes` and regular expression `test` method?